core/common/src/main/java/alluxio/worker/block/io/WorkerInternalBlockWorker.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.io;
+
+import alluxio.exception.BlockAlreadyExistsException;
+import alluxio.exception.BlockDoesNotExistException;
+import alluxio.exception.InvalidWorkerStateException;
+import alluxio.exception.WorkerOutOfSpaceException;
+import alluxio.grpc.AsyncCacheRequest;
+import alluxio.wire.BlockReadRequest;
+
+import java.io.IOException;
+
+/**
+ * An interface for worker internal clients to interact with worker operations directly
+ * without going through external RPC frameworks.
+ */
+public interface WorkerInternalBlockWorker {
+
+  /**
+   * Submits the async cache request to the worker to execute.
+   *
+   * @param request the async cache request
+   */
+  void asyncCache(AsyncCacheRequest request);
```

Comment:
when client is reading from a worker at the same process, we don't need to call `asyncCache` anymore. It will be a redundant operation b/c worker will have already get the data cached. Async cache operation is only needed if client on node A is reading from worker B and now wants the worker on the same node A to cache this block

Reply:
Yeah, the client on node A is reading from worker B and now wants the embedded worker A to cache this block? I think the above case is still valid?